src/main/java/com/ioet/bpm/skills/domain/exception/EntityNotFoundException.java

```diff
@@ -0,0 +1,7 @@
+package com.ioet.bpm.skills.domain.exception;
+
+public class EntityNotFoundException extends RuntimeException{
+    public EntityNotFoundException(String errorMessage){
```

Comment:
A not found row is not an error, it only indicates it was not found. I don't like to add exceptions for that.

Reply:
Gotcha!!. So what is the approach we are following here?

Reply:
If it will raise a response error (I'm not sure how the implementation works) I think is ok to launch an exception with a 404 error to the frontend.

Reply:
@fabricioflores This implementation follows this workflow:
1. The users of the endpoint send an ID and an object containing the skill's values to update.
2. The controller receives this request and it relies on the SkillService to update the Skill.
3. The service receives the ID and the skill's values and it updates the skill in the database.
4. If the update was successful, then the controller returns a `HttpStatus.OK` and the updated skill as part of the response body. On the other hand, if the ID is not found in the database, then the service returns an exception to the controller and the controller returns a HttpStatus.NOT_FOUND response. 
 
The reason I created the `EntityNotFoundException` was to notify the controller that the service could not update because the provided ID was not found. Doing in this makes sense to me because when you are updating a Skill, it is supposed that the provided ID must exist in the DB. If the provided ID does not exist, it is an atypical/exceptional case. 
It would be different if we were talking about a `search` method for instance. In this case, getting empty responses from the database is not an exceptional case because it is possible that search params might return empty results.

Reply:
So, maybe you can return a 422 if you don't want to return 404. But I think in this case 404 is more accurate about the error.